src/script/interpreter.cpp

```diff
@@ -1634,6 +1737,30 @@ static bool ExecuteWitnessScript(const Span<const valtype>& stack_span, const CS
 {
     std::vector<valtype> stack{stack_span.begin(), stack_span.end()};
 
+    if (sigversion == SigVersion::TAPSCRIPT) {
+        // OP_SUCCESSx processing overrides everything, including stack element size limits
+        CScript::const_iterator pc = scriptPubKey.begin();
+        while (pc < scriptPubKey.end()) {
+            opcodetype opcode;
+            if (!scriptPubKey.GetOp(pc, opcode)) {
+                // Note how this condition would not be reached if an unknown OP_SUCCESSx was found
```

Comment:
But only if `OP_SUCCESSx` occurred earlier?  Is it worth the effort to check the whole script for `OP_SUCCESSx` before and then do second pass for other conditions?

Reply:
Nope, the spec is defined as first parsing the script into opcodes to check for OP_SUCCESSx, and then executing.

Reply:
so you should check for e.g. unexecuted branches as well.

Reply:
Afaik this code fails if it finds an unknown OPCODE _before_ OP_SUCCESSx, so then it doesn't match the spec?

Reply:
If by unknown opcode you mean undecodable, no - because if an opcode can't be decoded, there are also no other opcodes decodable after it.

Reply:
It wasn't entirely clear to me what `GetScriptOp` does, in particular what happens if an `opcode` is does not occur in the `opcodetype` enum, but [apparently this is fine](https://stackoverflow.com/a/18195408).